### PR TITLE
Support custom images

### PR DESCRIPTION
**Pull request by aiyengar2**
This PR adds support for custom images via providing the resource's ARM resource identifier.

More info on ARM resource identifier format: https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.management.batch.models.imagereference.id?view=azure-dotnet#Microsoft_Azure_Management_Batch_Models_ImageReference_Id

Major updates:
- Introduced arm.go + customimage.go to create a unified way of referring to resources that are parsed from ARM resource identifiers

Misc. Updates:
- `getImageReference` now also pulls custom images / gallery images
- Added an error to `CreateVirtualMachine` if a user does not use managed disks with a custom image

Related Issue: https://github.com/rancher/rancher/issues/25342

Dependent on PR: https://github.com/rancher/machine/pull/72

### CODE REVIEW

**Comment by deniseschannon** [comment]
https://github.com/rancher/rancher/issues/23010

**Comment by aiyengar2** [comment]
After discussion with @dramich and @luthermonson offline, the code has been simplified. Tested it on Rancher and works as expected!

File: drivers/azure/azureutil/azureutil.go [hunk 1]
```diff
@@ -583,16 +583,22 @@ func (a AzureClient) CreateVirtualMachine(ctx context.Context, resourceGroup, na
...
 			Publisher: to.StringPtr(urn[0]),
 			Offer:     to.StringPtr(urn[1]),
 			Sku:       to.StringPtr(urn[2]),
 			Version:   to.StringPtr(urn[3]),
-		}
-		return imageReference, nil
+		}, nil
 	}
-	return nil, fmt.Errorf("image provided does not have a valid format: expected: <publisher>:<offer>:<sku>:<version>, got %s", image)
+	return nil, fmt.Errorf("image provided must be an image urn or an ARM resource identifier for a custom image / gallery image version")
```
Comment (dramich) [inline drivers/azure/azureutil/azureutil.go:601]: Can we update this to `image provided must be an image URN or an ARM resource identifier`

Comment (aiyengar2) [reply drivers/azure/azureutil/azureutil.go]: fixed in latest commit

File: Dockerfile.dapper [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:19.04
+FROM ubuntu:20.04
```
Comment (dramich) [inline Dockerfile.dapper:1]: This should be a separate PR

Comment (aiyengar2) [reply Dockerfile.dapper:1]: will be addressed in https://github.com/rancher/machine/pull/79
